src/main/java/uk/gov/hmcts/darts/task/service/impl/ManualDeletionProcessorImpl.java

```diff
@@ -0,0 +1,99 @@
+    private final ExternalUnstructuredDataStoreDeleter unstructuredDeleter;
+
+    @Override
+    public void process() {
+        OffsetDateTime deletionThreshold = getDeletionThreshold();
+        List<ObjectAdminActionEntity> actionsToDelete = objectAdminActionRepository.findFilesForManualDeletion(deletionThreshold);
+
+        for (ObjectAdminActionEntity objectAdminAction : actionsToDelete) {
+            if (isMediaNotDeleted(objectAdminAction)) {
+                deleteMedia(objectAdminAction.getMedia());
+            } else if (isTranscriptionNotDeleted(objectAdminAction)) {
+                deleteTranscriptionDocument(objectAdminAction.getTranscriptionDocument());
+            }
+        }
+
+    }
+
+    private void deleteMedia(MediaEntity mediaEntity) {
+
+        externalObjectDirectoryRepository.findByMediaStatusAndLocation(mediaEntity.getId())
+            .stream()
+            .peek(this::deleteFromExternalDataStore)
+            .forEach(externalObjectDirectoryRepository::delete);
+
+        mediaEntity.setDeleted(true);
+        mediaRepository.save(mediaEntity);
+        log.info("Deleted mediaEntity with ID: {}", mediaEntity.getId());
+    }
+
+    private void deleteTranscriptionDocument(TranscriptionDocumentEntity transcription) {
+        externalObjectDirectoryRepository.findByTranscriptionStatusAndLocation(transcription.getId())
+            .stream()
+            .peek(this::deleteFromExternalDataStore)
+            .forEach(externalObjectDirectoryRepository::delete);
+
+        transcription.setDeleted(true);
+        transcriptionDocumentRepository.save(transcription);
+        log.info("Deleted transcription document with ID: {}", transcription.getId());
+    }
+
+    private void deleteFromExternalDataStore(ExternalObjectDirectoryEntity externalObjectDirectoryEntity) {
+        if (ExternalLocationTypeEnum.INBOUND.getId().equals(externalObjectDirectoryEntity.getExternalLocationType().getId())) {
+            inboundDeleter.delete(externalObjectDirectoryEntity);
+        } else if (ExternalLocationTypeEnum.UNSTRUCTURED.getId().equals(externalObjectDirectoryEntity.getExternalLocationType().getId())) {
+            unstructuredDeleter.delete(externalObjectDirectoryEntity);
+        } else {
+            log.error("Can only delete INBOUND (1) and UNSTRUCTURED (2) data but tried to delete : {}",
+                      externalObjectDirectoryEntity.getExternalLocationType().getId());
+        }
+    }
+
+    private boolean isTranscriptionNotDeleted(ObjectAdminActionEntity action) {
+        return action.getTranscriptionDocument() != null && !action.getTranscriptionDocument().isDeleted();
+    }
+
+    private boolean isMediaNotDeleted(ObjectAdminActionEntity action) {
+        return action.getMedia() != null && !action.getMedia().isDeleted();
+    }
+
+    public OffsetDateTime getDeletionThreshold() {
+        return OffsetDateTime.now().minus(Duration.parse("PT" + gracePeriod));
```

Comment:
in case its helpful, we have a `CurrentTimeHelper` class that we often use instead of calling ffsetDateTime.now(), this allows us to mock the time in tests.
No need to change if you don't want to.

Reply:
I will leave this as it is.